webfront/views/cache.py

```diff
@@ -6,11 +6,66 @@
 
 from rest_framework import status
 from webfront.response import Response
+from webfront.views.utils import endpoints
+
 from django.conf import settings
 
 multiple_slashes = re.compile("/+")
 
 
+FIVE_DAYS = 5 * 24 * 60 * 60
+SHOULD_NO_CACHE = -1
+
+names = [ep["name"].lower() for ep in endpoints]
+
+
+def get_timeout_from_path(path, endpoint_levels):
+    parsed = urlparse(path)
+    # process query
+    query = parse_qs(parsed.query, keep_blank_values=True)
+
+    if (  # is requesting by accession
+        len(endpoint_levels) == 3
+        and len([ep for ep in endpoint_levels if ep in names]) < 2
+    ):
+        # it doesn't have modifiers
+        if len(query.keys()) == 0:
+            return SHOULD_NO_CACHE
+        if (  # The only modifier is page_size=20 or default
+            len(query.keys()) == 1
+            and "page_size" in query
+            and query["page_size"]
+            == settings.INTERPRO_CONFIG.get("default_page_size", 20)
+        ):
+            return SHOULD_NO_CACHE
+
+    # order querystring, lowercase keys
+    query = OrderedDict(
+        sorted(((key.lower(), sorted(value)) for key, value in query.items()))
+    )
+    page = query.get("page")
+    if page is not None and len(page) > 0:
+        try:
+            page = int(page[0])
+            if page > 1:
+                return FIVE_DAYS
+        except Exception:
+            return 0
```

Comment:
According to [this](https://docs.djangoproject.com/en/3.2/topics/cache/#basic-usage), a timeout of `0` won't cache the value. For consistency, couldn't we return `SHOULD_NO_CACHE`?

Reply:
> According to [this](https://docs.djangoproject.com/en/3.2/topics/cache/#basic-usage), a timeout of `0` won't cache the value. For consistency, couldn't we return `SHOULD_NO_CACHE`?

Nope. `0` means cache indefinitely... I'll create a constant to make it clearer,

Reply:
OK ignore the previous messages, you are right, that is a `SHOULD_NO_CACHE`. It seems I can't read my own code 🤦🏽‍♂️